packages/graphql/tests/schema/directives/selectable.test.ts

```diff
@@ -0,0 +1,717 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { printSchemaWithDirectives } from "@graphql-tools/utils";
+import { lexicographicSortSchema } from "graphql/utilities";
+import { gql } from "graphql-tag";
+import { Neo4jGraphQL } from "../../../src";
+import { TestSubscriptionsMechanism } from "../../utils/TestSubscriptionsMechanism";
+
+describe("@selectable", () => {
+    let subscriptionPlugin: TestSubscriptionsMechanism;
+
+    beforeAll(() => {
+        subscriptionPlugin = new TestSubscriptionsMechanism();
+    });
+
+    test("Disable read fields", async () => {
+        const typeDefs = gql`
+            type Movie {
+                title: String!
+                description: String @selectable(onRead: false, onAggregate: true)
+            }
+        `;
+        const neoSchema = new Neo4jGraphQL({ typeDefs });
+        const printedSchema = printSchemaWithDirectives(lexicographicSortSchema(await neoSchema.getSchema()));
```

Comment:
Is it worth updating the tests to use the new style of assertions for schema tests such as here? https://github.com/neo4j/graphql/blob/02909bb18fa482a196072d24c3ed39080dbb2e68/packages/graphql/tests/schema/directives/query.test.ts#L26

Reply:
It is definitely worth it at some point, however, the approach taken in `query.test.ts` is quite naive, as it only checks for top-level existence of fields, while this test needs to check for fields. Parsing the graphql tree in our tests directly is going to be a recipe for complex and hard testing. We need to find a better approach for this IMO. So if that's ok, I'll leave these tests as they are and make a point to discuss this in the future